test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -98,25 +96,17 @@ static bool stubSend( AgentMessageContext_t * pMsgCtx,
  * @brief A mocked receive function for the agent to receive commands.
  */
 static bool stubReceive( AgentMessageContext_t * pMsgCtx,
-                         void * pBuffer,
+                         Command_t ** pReceivedCommand,
                          uint32_t blockTimeMs )
 {
-    Command_t ** pCommandToReceive = ( Command_t ** ) pBuffer;
-
     ( void ) blockTimeMs;
 
     if( receiveCounter == 0 )
     {
-        *pCommandToReceive = pMsgCtx->pSentCommand;
-        receiveCounter++;
-        return true;
-    }
-    else
-    {
-        *pCommandToReceive = NULL;
-        receiveCounter++;
-        return false;
+        *pReceivedCommand = pMsgCtx->pSentCommand;
     }
+
+    return( receiveCounter++ == 0 );
```

Comment:
Still prefer that the return value be explicitly set in the block for `receiveCounter == 0` when it already exists. If the block wouldn't have existed, I would have been fine with the expression in the return statement

Reply:
Changede